caillef/dustzh.lua

```diff
@@ -659,20 +708,26 @@ weaponsMetatable = {
             hitMarker:hide()
             self.hitMarker = hitMarker
 
-            local ammoCount = ui:createText("20/20", Color.White, "big")
+            local weaponName = ui:createText("weaponsName", Color.Black)
             self.ammoCountText = ammoCount
-			ammoCount.parentDidResize = function(self)
-                self.pos = { Screen.Width * 0.5 - self.Width * 0.5, 100 - self.Height - 10, 0 }
+            self.weaponNameText = weaponName
+            weaponName.parentDidResize = function(self)
+                self.pos = { Screen.Width / 25 - weaponName.Width / 1.75, Screen.Height / 3, 0 }
             end
-            ammoCount:parentDidResize()
+            weaponName:parentDidResize()
             self:updateAmmoUI()
+            self:updateNameUI()
 
 			self.templates = {}
         end,
         updateAmmoUI = function(self)
             if self.ammo == nil then return end
-            self.ammoCountText.Text = math.floor(self.ammo).."/"..self.maxAmmo
-			self.ammoCountText.pos = { Screen.Width * 0.5 - self.ammoCountText.Width * 0.5, 100 - self.ammoCountText.Height - 10, 0 }
+            addAmmoIndication(math.floor(self.ammo))
+        end,
+        updateNameUI = function(self)
+            if self.weaponName == nil then return end
+            self.weaponNameText.Text = self.weaponName
+            self.weaponNameText.pos = { Screen.Width / 25 - self.weaponNameText.Width / 1.75, Screen.Height / 3, 0 }
```

Comment:
`parentDidResize` could be called instead of duplicating position logic. 
By the way, with that logic, when the screen is too small, the weapons name is displayed out of screen. 

And I think the UI components in this area (ammo, name, life) should always be updated all together.

Reply:
<img width="272" alt="Screenshot 2023-07-21 at 08 08 23" src="https://github.com/cubzh/worlds/assets/6775074/853616e6-3088-48c9-94b0-bfadb01d250d">

<img width="209" alt="Screenshot 2023-07-21 at 08 09 10" src="https://github.com/cubzh/worlds/assets/6775074/a9e0459f-c159-4325-b778-480598780609">